### DISCUSSION TITLE
Customize the number of typos

### DISCUSSION THREAD

**Discussion by devig** [meilisearch/product#300] [Feedback & Feature Proposal]
Customize the number of typos allowed.
Examples:
0 - disable typo tolerance.
2 - allow two typos.

And add option to allow typo(s) on synonyms.

**Comment by curquiza** [comment]
Hello @devig 
Thanks for your suggestion!

Not exactly the same, but related to https://github.com/meilisearch/product/discussions/176

Can we share with us your use case? Also, how and in which context you would need this feature? 🙂
Thanks!

**Reply by devig** [reply]
#176 it's only about disable or enable... I left proposal about customization of the number of typos (that's more flexible).

Sometimes people make typos and it's really important for business to be in tune and no matter what it takes show search results then show nothing. 
I tell you more even English language for different countries allow different variations (take a look https://upload.wikimedia.org/wikipedia/commons/a/a2/International_English_Spelling.png).
A lot of not native speakers who don't remember how exactly write some words - it's the third argument.

Allowing numbers of typos should be because:
Example: in db you have word "traveling". I am from Canada and "travelling" it's right for me and I do one typo "travalling" and see nothing found...(
Or just "goose" and "geese".  

There a lot of more complex examples with articles (sku) of products or other languages that a more complex when English and sometimes should have possibility of two typos.

I am in a long way of searching worthy alternative of Elasticsearch and Sphinx for the full text fuzzy search system with synonyms. 
Every customer (that I met) who wished the search system added points about possibility of more then one typo and  synonyms and typos in synonyms. And for now it's the last reason why I must use Elasticsearch or Sphinx instead of Meilisearch.

I understand that two typos and typos in synonyms could increase a memory usage or be reason showing not relevant results, but it's better to have customization than not to have and to use other systems. By default you could do typos:0

I hope you got my point. And you see the prospects for development Meilisearch.

**Reply by gmourier** [reply]
Hello @devig! I'm locking this discussion to favor #312 which talks about further customization of the behavior for the typo tolerance feature.

**Comment by gmourier** [comment]
Locked in favor of #312.
